File: 3869-smallest-index-with-digit-sum-equal-to-index/smallest-index-with-digit-sum-equal-to-index.java
class Solution {
    public int smallestIndex(int[] nums) {
        for(int j=0;j<nums.length;j++){
            int n=nums[j];
            int sum=0;
            while(n!=0){
                sum+=n%10;
                n/=10;
            }
            if(j==sum){
                return j;
            }
        }
        return -1;
    }
}

File: 3869-smallest-index-with-digit-sum-equal-to-index/smallest-index-with-digit-sum-equal-to-index.cpp
class Solution {
public:
    int smallestIndex(vector<int>& arr) {
        for(int i=0;i<arr.size();i++){
            if(i==sum(arr[i])){
                return i;
            }
        }
        return -1;
    }
    int sum(int n){
        // while(n>10){
        int sum=0;
            while(n!=0){
                int digit=n%10;
                sum+=digit;  
                n/=10;    
            }
        //     n=sum;
        // }
        return sum;
    }
};